components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/java/org/wso2/carbon/mediator/enrich/ui/EnrichMediator.java

```diff
@@ -177,13 +178,23 @@ public OMElement serialize(OMElement parent) {
 
             if (sourceInlineXML != null && !sourceInlineXML.equals("")) {
                 OMElement inlineXMLElement;
+                JsonParser parser = new JsonParser();
+                // check for inline JSON values
                 try {
-                    inlineXMLElement = AXIOMUtil.stringToOM(sourceInlineXML);
-                    if (inlineXMLElement != null) {
-                        sourceElem.addChild(inlineXMLElement);
+                    parser.parse(sourceInlineXML);
+                    String formattedString = "<source xmlns=\"" + SynapseConstants.SYNAPSE_NAMESPACE +
+                            "\" type=\"inline\">" + sourceInlineXML + "</source>";
+                    sourceElem = AXIOMUtil.stringToOM(formattedString);
+                } catch (Exception ex) {
+                    // not a JSON
+                    try {
+                        inlineXMLElement = AXIOMUtil.stringToOM(sourceInlineXML);
+                        if (inlineXMLElement != null) {
+                            sourceElem.addChild(inlineXMLElement);
+                        }
+                    } catch (XMLStreamException e) {
+                        e.printStackTrace();
```

Comment:
Don't we miss the exception information in log files if we use `e.printStackTrace();`?

Reply:
This was there in the old code. I just change the order. I can't see any logging mechanism in any of the mediator codes in org.wso2.carbon.mediator package

Reply:
+1 fixed